extensions/amp-date-picker/0.1/test/test-amp-date-picker.js

```diff
@@ -167,5 +171,141 @@ describes.realWin('amp-date-picker', {
+      it('should not create hidden inputs outside of forms', () => {
+        const {element} = createDatePicker();
+
+        expect(element.querySelector('input[type="hidden"]')).to.be.null;
+      });
+
+      it('should create a hidden input when inside a form', () => {
+        const form = document.createElement('form');
+        document.body.appendChild(form);
+        const {element, layoutCallback} = createDatePicker({}, form);
+
+        return layoutCallback().then(() => {
+          const input = element.querySelector('input[type="hidden"]');
+          expect(input).to.not.be.null;
+          expect(input.name).to.equal('date');
+        });
+      });
+
+      it('should name the input `${id}-date` when another ' +
+          '#date input exists', () => {
+        const form = document.createElement('form');
+        const dateInput = document.createElement('input');
+        dateInput.type = 'hidden';
+        dateInput.name = 'date';
+        form.appendChild(dateInput);
+        document.body.appendChild(form);
+        const {element, layoutCallback} =
+            createDatePicker({id: 'delivery'}, form);
+
+        return layoutCallback().then(() => {
+          const input = element.querySelector('input[type="hidden"]');
+          expect(input).to.not.be.null;
+          expect(input.name).to.equal('delivery-date');
+        });
+      });
+
+      it('should error if the input when another ' +
+          '#date input exists and the picker has no ID', () => {
+        const form = document.createElement('form');
+        const dateInput = document.createElement('input');
+        dateInput.type = 'hidden';
+        dateInput.name = 'date';
+        form.appendChild(dateInput);
+        document.body.appendChild(form);
+
+        const {layoutCallback} = createDatePicker({}, form);
+
+        allowConsoleError(() => {
+          expect(layoutCallback()).to.be.rejectedWith(
+              'another #date input exists');
+        });
+      });
+    });
+
+    describe('hidden inputs in range date picker in forms', () => {
+      it('should not create hidden inputs outside of forms', () => {
+        const {element} = createDatePicker({type: 'range'});
+
+        expect(element.querySelector('input[type="hidden"]')).to.be.null;
+      });
+
+      it('should create a hidden input when inside a form', () => {
+        const form = document.createElement('form');
+        document.body.appendChild(form);
+        const {element, layoutCallback} =
+            createDatePicker({type: 'range'}, form);
+
+        return layoutCallback().then(() => {
+          const inputs = element.querySelectorAll('input[type="hidden"]');
+          expect(inputs.length).to.equal(2);
+          expect(inputs[0].name).to.equal('start-date');
+          expect(inputs[1].name).to.equal('end-date');
+        });
+      });
+
+      it('should name one of the inputs `${id}-date` when another ' +
```

Comment:
Consider renaming to:
"should name one of the inputs `${id}-(start|end)-date` when another"

Reply:
Good call, I'll do that